docs/source/1.0/spec/core/traits.rst

```diff
@@ -191,6 +191,158 @@ to call ``CreateTable`` on a table that already exists will return an error.
+        update: UpdateForecast
+    }
+
+    operation UpdateForecast {
+       input: UpdateForecastInput
+    }
+
+    structure UpdateForecastInput {
+        chanceOfRain: Float
+
+        @notProperty
+        dryRun: Boolean
+
+        @idempotencyToken
+        clientToken: String
+    }
+
+
+.. smithy-trait:: smithy.api#property
+.. _property-trait:
+
+------------------
+``property`` trait
+------------------
+
+Summary
+    Binds a top-level input or output shape member of a structure to a resource
+    property with a different name.
+Trait selector
+    ``:is(operation -[input, output]-> structure > member)``
+
+    *A top-level member of an operation's input or output shape*
+Value type
+    An object with the following properties:
+
+    .. list-table::
+       :header-rows: 1
+       :widths: 10 23 67
+
+       * - Property
+         - Type
+         - Description
+       * - name
+         - ``string``
+         - Name of the resource property to bind the member to. This field MUST
+           be specified.
+
+By default, top-level input or output shape members are bound to the resource
+property with the same name. In situations where this isn't possible, this
+trait can be used to specify which property the member is bound to.
+
+   .. admonition:: Note
+    :class: tip
+
+    This trait should only be used for existing service APIs that need to
+    maintain backwards compatibility with input and output structures, while
+    enabling Smithy's resource property modeling and validation.
+
+.. code-tab:: smithy
+
+    resource Forecast {
+        properties: { chanceOfRain: Float }
+        read: GetForecast
+    }
+
+    @readonly
+    operation GetForecast {
+       output: GetForecastOutput
+    }
+
+    structure GetForecastOutput {
+        @property(name: "chanceOfRain")
+        howLikelyToRain: Float
+    }
+
+
+.. smithy-trait:: smithy.api#nestedProperties
+.. _nested-properties-trait:
+
+--------------------------
+``nestedProperties`` trait
+--------------------------
+
+Summary
+    Allows the binding of resource properties to occur with a members nested
+    deeper than the lifecycle operation's input or output shape.
+Trait selector
+    ``:is(operation -[input, output]-> structure > member)``
+
+    *A member of a top-level input or output structure member*
+Value type
+    Annotation trait.
+
+By default, top-level input and output shape members of lifecycle operations are
+bound to resource properties. If the shape containing the top-level resource
+properties is nested one level deeper than the input and output shape members,
+apply this trait to that member.
+
+.. code-block:: smithy
+
+    resource Forecast {
+        properties: { chanceOfRain: Float }
+        read: GetForecast
+    }
+
+    @readonly
+    operation GetForecast {
+       output: GetForecastOutput
+    }
+
+    structure GetForecastOutput {
+        @nestedProperties
+        forecastData: ForecastData
+    }
+
+    structure ForecastData {
```

Comment:
Would it be useful to use type elision syntax and the new `for Forecast` syntax here? That's what I think I'd expect people to do in practice. The only catch is you probably want to link to the `for` syntax addition from here using a `.. seealso` directive.